src/FondOfSpryker/Yves/GoogleTagManageQuoteConnector/Provider/GoogleTagManageQuoteProviderInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManageQuoteConnector\Provider;
+
+interface GoogleTagManageQuoteProviderInterface
+{
+    /**
+     * @param string $page
+     * @param array $params
+     *
+     * @return array
+     */
+    public function addVariable(string $page, array $params): array;
+}
```

Comment:
This should be the general plugin interface instead of it's own interface, so `GoogleTagManageQuoteProvider` should implement `GoogleTagManagerVariableBuilderPluginInterface`. It's an enhanced plugin which combines multiple plugins and injects the `QuoteClient`.

Reply:
Mhhhh, but `GoogleTagManageQuoteProvider` is not Plugin. of course i can implement it, but i am not sure if it is correct. 

the point is we dont want logic inside plugin/s. there is also an `GoogleTagManageQuoteProviderPlugin` which only creates `GoogleTagManageQuoteProvider` by factory  and call `addVariable(string $page, array $params)` I am also not really happy with the methodname `addVariable`, in my opinion `attachQuoteToVariablePlugins` or `provideQuoteToVariablePlugins` sounds better.